### PR TITLE
[SYCL] Balance device reference counter

### PR DESCRIPTION
**Pull request by romanovvlad**
While it's not strictly required to release(decrement reference counters)
of handles of root devices it might be convenient for plugins to have
the same behavior as for other handles.

### CODE REVIEW

**Comment by romanovvlad** [comment]
Related github issue https://github.com/intel/llvm/issues/6034

**Comment by romanovvlad** [comment]
@smaslov-intel Could you please review?

**Review by smaslov-intel — APPROVED** [review]

**Review by smaslov-intel — APPROVED** [review]
LGTM

File: sycl/unittests/CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,6 @@ add_subdirectory(Extensions)
 add_subdirectory(windows)
 add_subdirectory(event)
 add_subdirectory(buffer)
-add_subdirectory(context)
+add_subdirectory(context_device)
```
Comment (bader) [inline sycl/unittests/CMakeLists.txt:49]: Looks like `context_device` directory is missing in this PR.

Comment (romanovvlad) [reply sycl/unittests/CMakeLists.txt:49]: Fixed.

**Comment by romanovvlad** [comment]
@v-klochkov  Could you please take a look?

**Review by sergey-semenov — APPROVED** [review]

**Pull request merged**